### ISSUE TITLE
change parsing method

### ISSUE THREAD

**Issue by wvlia5** [Torrencem/PyAPL#4]
wouldn't it be "better" (simpler code) to do the parsing in such a way (recursive descent parsing?) that you can compute the results of the more deeply nested parentheses first and going up recursively, instead of just computing everything from right to left? maybe what I say wouldn't work for a case such as  "(a+5) ⍴ a←3"? pheraps computing everything from right to left, until you hit a nested expression, computing it recursively, and continuing?

**Comment by Torrencem** [comment]
Yes, this is exactly right! I knew about this as I was righting the code initially. I will have to change how that works
